beamer/chain.py

```diff
@@ -49,8 +49,8 @@ def __init__(
         contracts: tuple[Contract, ...],
         deployment_block: BlockNumber,
         on_new_events: list[Callable[[list[Event]], None]],
-        on_sync_done: Callable[[], None],
-        poll_period: int,
+        on_sync_done: list[Callable[[], None]],
```

Comment:
I would suggest making a separate type for the callable, e.g.

```python
SyncDoneCallback = Callable[[], None]
```
which would then allow us to be have readable

```python
        on_sync_done: list[SyncDoneCallback],
```
(not sure whether the type needs to be private, but if it can be made, then we should make it private).